File: javascript/leetcode/1413. Minimum Value to Get Positive Step by Step Sum.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minStartValue = function(nums) {
    let total = 0, min_total = 0;
    
    for(let i of nums) {
        total += i;
        min_total = Math.min(min_total, total)
    }
    
    return 1 - min_total
};

File: c++/leetcode/1413. Minimum Value to Get Positive Step by Step Sum.cpp
class Solution {
public:
    int minStartValue(vector<int>& nums) {
        auto total = 0, min_total = 0;
        
        for(auto i : nums) {
            total += i;
            min_total = min(min_total, total);
        }
        
        return 1 - min_total;
    }
}; 
